provider/ec2/environ.go

```diff
@@ -1546,19 +1553,21 @@ func (e *environ) AllInstancesByState(ctx context.ProviderCallContext, states ..
 	} else if err != nil {
 		return nil, errors.Trace(maybeConvertCredentialError(err, ctx))
 	}
-	filter := ec2.NewFilter()
-	filter.Add("instance-state-name", states...)
-	filter.Add("instance.group-id", group.Id)
-	return e.allInstances(ctx, filter)
+	filters := []*sdkec2.Filter{
+		makeFilter("instance-state-name", states...),
+		makeFilter("instance.group-id", group.Id),
+	}
+	return e.allInstances(ctx, filters)
 }
 
 // ControllerInstances is part of the environs.Environ interface.
 func (e *environ) ControllerInstances(ctx context.ProviderCallContext, controllerUUID string) ([]instance.Id, error) {
```

Comment:
This is only used in tests IIANM.  Most of those are JujuConnsuite which means the dummy provider is in use (the sooner that dies in a big fire the better). The one relevant ec2 test is TestBootstrapAndDeploy which is on a LiveTests suite - this suite will use a real AWS if run with credentials.
Just mentioning it as it would be worth considering how best to remove this method off the provider given it doesn't seem to earn its keep in terms of being maintained etc if not used in production.

Reply:
Yeah, doing a quick search and it does look like that's the case (it's only used in tests). I'll keep it in mind; doesn't seem in scope for this PR though.